src/core/shred.zig

```diff
@@ -11,4 +12,52 @@ pub const ShredVersion = struct {
     pub fn init_manually_set(version: u16) Self {
         return Self{ .value = version };
     }
+
+    pub fn version_from_hash(hash: *Hash) u16 {
+        const hash_bytes = hash.data;
+        var accum: [2]u8 = .{ 0, 0 };
+        var chunks = std.mem.window(u8, &hash_bytes, 2, 2);
+
+        while (chunks.next()) |chunk| {
+            accum[0] ^= chunk[0];
+            accum[1] ^= chunk[1];
+        }
+        var version = (@as(u16, accum[0]) << 8) | accum[1];
+        return version +| 1;
+    }
+
+    pub fn compute_shred_version(genesis_hash: Hash, maybe_hard_forks: ?HardForks, alloc: Allocator) u16 {
```

Comment:
(small nit) if `Allocator` is a fcn parameter, can we make it the first parameter